plugins/generator-1.16.5/forge-1.16.5/templates/tool.java.ftl

```diff
@@ -356,6 +362,40 @@ public class ${name}Item extends ${JavaModName}Elements.ModElement{
 			return ${data.enchantability};
 		}
 
+	}
+<#elseif data.toolType == "Fishing rod">
+    private static class ItemToolCustom extends FishingRodItem {
+
+		protected ItemToolCustom() {
+			super(new Item.Properties()
+				.group(${data.creativeTab})
+				.maxDamage(${data.usageCount})
+				<#if data.immuneToFire>
+				.isImmuneToFire()
+				</#if>
+			);
+		}
+
+		@Override public boolean hitEntity(ItemStack stack, LivingEntity target, LivingEntity attacker) {
+			stack.damageItem(2, attacker, i -> i.sendBreakAnimation(EquipmentSlotType.MAINHAND));
+			return true;
+		}
+
+		@Override public int getItemEnchantability() {
+			return ${data.enchantability};
+		}
+
+		<#if data.repairItems?has_content>
+		    @Override
+            public boolean getIsRepairable(ItemStack toRepair, ItemStack repair) {
+                <#list data.repairItems as repairItem>

```

Comment:
I tried to repair a fishing rod, but it didn't work.
This template should fix the issue (we get the item from `repair`, then compare it to the selected items)
```
Item repairItem = repair.getItem();
return
<#list data.repairItems as repairItem>
	repairItem == ${mappedMCItemToItem(repairItem)}
	<#if repairItem?has_next>||</#if>
</#list>;
```